tests/containers/search/filters/keyword/KeywordContainer-test.jsx

```diff
@@ -8,79 +8,82 @@ import { shallow } from 'enzyme';
-                    updateTextSearchInput={mockReduxActionKeyword} />);
+                    updateTextSearchInput={mockReduxAction} />);
 
-            const submitTextSpy = sinon.spy(keywordContainer.instance(),
-                'submitText');
+            const toggleKeywordSpy = sinon.spy(keywordContainer.instance(),
+                'toggleKeyword');
 
-            // Add keyword to redux
-            keywordContainer.setState({
-                value: 'Education'
-            });
-            keywordContainer.instance().submitText();
+            // Add Keyword to redux
+            keywordContainer.instance().toggleKeyword(keyword);
 
-            // everything should be updated now
-            expect(submitTextSpy.callCount).toEqual(1);
-            expect(mockReduxActionKeyword).toHaveBeenCalled();
+            // Everything should be updated now
+            expect(toggleKeywordSpy.callCount).toEqual(1);
+            expect(mockReduxAction).toHaveBeenCalled();
 
-            // reset the spies
-            submitTextSpy.reset();
+            // Reset the spy
+            toggleKeywordSpy.reset();
         });
-        it('should overwrite a previous keyword with a new keyword', () => {
-            const existingFilters = Object.assign({}, initialFilters, {
-                keyword: 'Education'
-            });
-            const mockReduxActionKeyword = jest.fn((args) => {
-                expect(args).toEqual('Financial');
+        it('should remove a Keyword that has been deselected from Redux', () => {
+            const mockReduxAction = jest.fn((args) => {
+                expect(args).toEqual("testing");
             });
+
+            // Set up container with mocked action
             const keywordContainer = shallow(
                 <KeywordContainer
-                    {...existingFilters}
-                    updateTextSearchInput={mockReduxActionKeyword} />);
+                    {...initialFilters}
+                    updateTextSearchInput={mockReduxAction} />);
+
+            const toggleKeywordSpy = sinon.spy(keywordContainer.instance(),
+                'toggleKeyword');
 
-            const submitTextSpy = sinon.spy(keywordContainer.instance(),
-                'submitText');
+            // Add Keyword to redux
+            keywordContainer.instance().toggleKeyword(keyword);
 
-            // Add keyword to redux
-            keywordContainer.instance().populateInput('Financial');
-            keywordContainer.instance().submitText();
+            // Remove Keyword from Redux
+            keywordContainer.instance().toggleKeyword(keyword);
 
-            // everything should be updated now
-            expect(submitTextSpy.callCount).toEqual(1);
-            expect(mockReduxActionKeyword).toHaveBeenCalled();
+            // Everything should be updated now
+            expect(toggleKeywordSpy.callCount).toEqual(2);
+            expect(mockReduxAction).toHaveBeenCalledTimes(2);
```

Comment:
How do these expectations prove the premise of "[container] should remove a Keyword that has been deselected from Redux"?